django/db/backends/base/schema.py

```diff
@@ -1246,14 +1250,15 @@ def create_unique_name(*args, **kwargs):
             name = IndexName(table, columns, '_uniq', create_unique_name)
         else:
             name = self.quote_name(name)
-        if condition or include or opclasses or expressions:
+        if condition or include or opclasses or expressions or db_tablespace:
             sql = self.sql_create_unique_index
         else:
             sql = self.sql_create_unique
         if columns:
             columns = self._index_columns(table, columns, col_suffixes=(), opclasses=opclasses)
         else:
             columns = Expressions(table, expressions, compiler, self.quote_value)
+        tablespace_sql = self._get_index_tablespace_sql(model, fields=[], db_tablespace=db_tablespace)
```

Comment:
I passed `fields` an empty list here as we don't have access to the actual field instances but instead. Felt that this might be OK since the auto-mapping of tablespaces fields and unique constraints never've been in place. If anyone thinks otherwise it looks like we could drop the `columns` argument to `_create_unique_sql` and add `fields` with actual field instances but not really sure how that would affect third-party backends.

Reply:
Yeah it's quite inconsistent :disappointed:. I think we should change `columns` to `fields` in a separate PR.

Reply:
Created https://github.com/django/django/pull/14459